buildSrc/build.gradle.kts

```diff
@@ -1,7 +1,40 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+  `java-library`
+  `kotlin-dsl`
+  alias(libs.plugins.jetbrains.kotlin)
+}
+
 repositories {
+  mavenLocal()
   mavenCentral()
+  gradlePluginPortal()
 }
 
 dependencies {
+  implementation(gradleApi())
+  implementation(libs.aspectj.postCompileWeaving)
   implementation(libs.jsoup)
+  implementation(libs.jetbrains.intellij)
+  implementation(libs.jetbrains.kotlin)
+  implementation(libs.spotless)
+  implementation(libs.checkerFramework)
+}
+
+val javaMajorVersion = JavaVersion.VERSION_11
+
+project.tasks.withType<KotlinCompile> {
+//  TODO (#785): revert this setting
+//  kotlinOptions.allWarningsAsErrors = true
```

Comment:
Here it can be uncommented, probably... it was only commented out due to some warnings in frontend code

Reply:
Unfortunately I had to use some gradle API code marked with deprecated. Additionally we have an unsafe typecast in one function due to type erasure, so that part will need a small refactor later.

Reply:
Okay... so pls open another issue, as #785 is unrelated to this place

Reply:
Updated #1004 accordingly, resolving